site/blocks/StickyNav.tsx

```diff
@@ -0,0 +1,202 @@
+import React, { createRef } from "react"
+import ReactDOM from "react-dom"
+import classnames from "classnames"
+import { throttle } from "../../clientUtils/Util.js"
+
+function getTotalOffset(element: HTMLElement): {
+    x: number
+    y: number
+} {
+    let y = 0
+    let x = 0
+    while (true) {
+        x += element.offsetLeft
+        y += element.offsetTop
+        if (!element.offsetParent) {
+            break
+        }
+        element = element.offsetParent as HTMLElement
+    }
+    return { x, y }
+}
+
+const STICKY_NAV_HEIGHT = 54
+const HEADING_SCROLL_MARGIN = 16
+
+interface StickyNavProps {
+    links: { text: string; target: string }[]
+}
+class StickyNav extends React.Component<
+    StickyNavProps,
+    {
+        headingPositions: { id: string; yPosition: number }[]
+        currentHeadingIndex?: number
+        links: { text: string; target: string }[]
```

Comment:
You are using `{ id: string; yPosition: number }` and `{ text: string; target: string }` many times around the code - consider pulling them out into a type (e.g. `StickyNavHeadingPosition` and `StickyNavLink`).